reflex/experimental/layout.py

```diff
@@ -0,0 +1,158 @@
+        """Create the sidebar component."""
+        direction = props.pop("direction", "left")
+        props.setdefault("border_right", f"1px solid {rx.color('accent', 12)}")
+        props.setdefault("background_color", rx.color("accent", 1))
+        props.setdefault("width", "20vw")
+        props.setdefault("height", "100vh")
+        return super().create(
+            drawer.trigger(
+                SidebarTrigger.create(), position="absolute", top="15", left="15"
+            ),
+            drawer.portal(
+                drawer.content(
+                    *children,
+                    **props,
+                )
+            ),
+            direction=direction,
+        )
+
+
+class SidebarTrigger(Fragment):
+    """A component that renders the sidebar trigger."""
+
+    @classmethod
+    def create(cls, sidebar: Component | None = None, **props):
+
+        trigger_props = {
+            **props,
+            "position": "absolute",
+            "z_index": "15",
+            "top": "15",
+            "background_color": "transparent",
+        }
+        if sidebar:
+            ...  # make stuff working with stateful sidebar
+            trigger_props["left"] = rx.cond(
+                sidebar.State.open, "calc(20vw - 32px)", "15"
+            )
+            trigger = rx.cond(
+                sidebar.State.open,
+                rx.icon_button(
+                    "arrow-left-from-line",
+                    on_click=sidebar.State.toggle,
+                    **trigger_props,
+                ),
+                rx.icon_button(
+                    "arrow-right-from-line",
+                    on_click=sidebar.State.toggle,
+                    **trigger_props,
+                ),
+            )
+        else:
+            ...  # make stuff working with stateless sidebar
+            trigger_props["left"] = "15"
+            return IconButton.create("arrow-right-from-line", **trigger_props)
+
+        return super().create(trigger)
+
+
+class Layout(Box):
+    """A component that renders the layout."""
+
+    @classmethod
+    def create(
+        cls,
+        content: Component,
+        sidebar: Component | None = None,
+        **props,
+    ):
+        """Create the layout component."""
+        layout_root = rx.hstack
+
+        if sidebar is None:
+            return rx.container(content, **props)
+
+        if isinstance(sidebar, Sidebar):
+            if sidebar.State:
+                ...  # make stuff working with stateful sidebar
+            else:
+                ...  # make stuff working with stateless sidebar
+        else:
+            sidebar = Sidebar.create(sidebar)
+        return layout_root(
+            sidebar,
+            # SidebarTrigger.create(
+            #     sidebar=sidebar_comp,
+            # ),
+            rx.container(
+                content,
+                height="100%",
+            ),
+            **props,
+            width="100vw",
+            height="100vh",
```

Comment:
this should probably be min_height, otherwise if page gets too long it can cause the app to no longer show the top of the page